src/linker/Linker.Steps/RootNonTrimmableAssemblies.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Mono.Linker.Steps
+{
+	public class RootNonTrimmableAssemblies : BaseStep
+	{
+		protected override void Process ()
+		{
+			// Walk over all -reference inputs and resolve any that may need to be rooted
+			foreach (var assemblyPath in GetInputAssemblyPaths ()) {
+				var assemblyName = Path.GetFileNameWithoutExtension (assemblyPath);
+
+				if (!MaybeIsFullyPreservedAssembly (assemblyName))
+					continue;
+
+				var assembly = Context.TryResolve (assemblyName);
+				if (assembly == null) {
+					Context.LogError ($"Reference assembly '{assemblyPath}' could not be loaded", 1039);
```

Comment:
I am wondering if this should be a warning instead. We shouldn't be passing in bad references in the first place, but the resolution logic here is stricter than what we used to do.

Reply:
Leaving it as an error for consistency with other command-line inputs. If it hurts people we can change to a warning.